src/main/scala/com/microsoft/hyperspace/index/IndexUtils.scala

```diff
@@ -55,4 +58,10 @@ object IndexUtils {
             s"from available source columns:\n${df.schema.treeString}")
     }
   }
+
+  /**
+   * Spark UDF to decode the URL-encoded string returned by input_file_name().
+   */
+  lazy val decodeInputFileName = udf(
+    (p: String) => URLDecoder.decode(p.replace("+", "%2B"), "UTF-8"))
```

Comment:
From comments in #482 
`input_file_name() returns a URL-encoded string and it doesn't match the file paths stored in the FileIdTracker if there are special characters in the path. CoveringIndex has the same issue and will be fixed in a separate PR.

The plus sign has to be treated specially because the built-in URLDecoder decodes "+" to " ".`